diff --git a/src/shared_text_stage.inc b/src/shared_text_stage.inc
index a1efa62..9433e52 100644
--- a/src/shared_text_stage.inc
+++ b/src/shared_text_stage.inc
@@ -544,7 +544,7 @@ stage_schedule
         fcb 2,8,4,TEXT_STAGE_NAME,0,0
         fdb $4D60
         fcb 2,12,4,TEXT_STAGE_NAME,0,0
-        fdb $4D84
+        fdb $5798
         fcb 1,0,3,TEXT_STAGE_PART,0,0
         fdb $618C
         fcb 3,0,4,TEXT_STAGE_BONUS,0,0
